modules/scrollback/index.js

```diff
@@ -110,108 +111,402 @@ module.exports.formatLine = function(line) {
+ * a deterministic specification of a flexible language for selecting text is
+ * defined based on three criteria: speaker name, regular expression match, and
+ * sequential number (indices in an inverse relationship with current time); then
+ * an algorithm for parsing the language specification and selecting the lines a
+ * user's instance of the specification matches is demonstrated.
+ */
+/* Get lines from scrollback using various range specifications
+ *
+ * There are three possible range specifiers: nick (N), regex (R), and line number (L).
+ * The formal grammar for these specifications:
+ *   All = Spec | All Spec | All, Spec
+ *   Spec = Nickopt Regexopt Lineopt
+ *   Nickopt = e | Nicks
+ *   Regexopt = e | Regexes
+ *   Lineopt = e | Lines
+ *   Nicks = N | Nicks,N
+ *   Regexes = R | Regexes,R
+ *   Lines = L | L-L | L..L | L...L | Lines,Lines
+ *
+ * Line parsing is handled by https://github.com/euank/node-parse-numeric-range
+ *
+ * In general, a range specification after being parsed by numeric-range looks like:
+ *   [N1[,N2][,N3]...[,Ni]] [R1[,R2][,R3]...[,Rj]] [L1[,L2][,L3]...[,Lk]][,]
+ * This parses to a cross-product of ranges:
+ *   [N1,R1,L1], [N1,R1,L2], ..., [N1,R1,Lk],
+ *   [N1,R2,L1], [N1,R2,L2], ..., [N1,R2,Lk],
+ *   ...
+ *   [N1,Rj,L1], ..
+ *   (and so on and so forth for N2,...,Ni).
+ *
+ * If no nick is specified, lines from every nick will be considered;
+ * if no regex is specified, every line will match;
+ * if no line is specified, the first line matching the range is returned.
+ *
+ * Multiple range specs may be concatenated together.
+ * In cases where ambiguity could occur, a trailing , can be used to indicate
+ * where one range ends and the next begins.
+ *
+ * Examples:
+ * joe /dog/ 3
+ * -> Return the third line from user joe matching /dog/
+ *
+ * joe /dog/,/cat/
+ * -> Return the first line from user joe matching /dog/,
+ *    then the first line from joe matching /cat/
+ *
+ * joe /dog/ /cat/
+ * -> Return the first line from user joe matching /dog/,
+ *    then the first line matching /cat/
+ *
+ * joe,bob 3
+ * -> Return the third line from joe, then the third line from bob.
+ *
+ * joe, /dog/, 3
+ * -> Return the first line from joe, then the first line matching /dog/,
+ *    then the third line.
+ *
+ * joe 3 bob /dog/ /cat/ bill, 4
+ * -> Return the third line from joe, then the first line from bob
+ *    matching /dog/, then the first line matching /cat/, then the
+ *    first line from bill, then the fourth line.
+ */
+var STATES = Object.freeze({
+  NICK: 1,
+  REGEX: 2,
+  LINES: 3,
+  NEW: 4,
+  ERROR: 5
+});
+
+module.exports.parseSpecs = function(input) {
+  var specs = [];
+  var spec = {nicks: [], regexes: [], lines: []};
+
+  // State machine variables
+  //var state = ''; // one of 'nick', 'regex', 'lines', 'new', 'error'
+  var error = '';
+
+  var i = 0;
+  var result = findNextState(input, i, state);
```

Comment:
`state` is used here but was commented out on line 196. Either explicitly pass in undefined or initialize state to undefined or something